### PR TITLE
[SMALLFIX] Deleting a file or directory should lock the parent directory

### PR DESCRIPTION
**Pull request by gpang**

### CODE REVIEW

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java [hunk 1]
```diff
@@ -611,6 +615,42 @@ public void ttlRename() throws Exception {
     Assert.assertEquals(ttl, folderInfo.getTtl());
   }
 
+  @Test
+  public void concurrentCreateDeleteTest() throws Exception {
+    List<Future<?>> futures = new ArrayList<>();
+    AlluxioURI directory = new AlluxioURI("/dir");
+    AlluxioURI[] files = new AlluxioURI[10];
+    final int numThreads = 8;
+    final int sleepMs = 3000;
```
Comment (aaudiber) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: maybe name this `testDurationMs`?

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java [hunk 2]
```diff
@@ -611,6 +615,42 @@ public void ttlRename() throws Exception {
...
+      for (Future<?> future : futures) {
+        future.get();
+      }
+
+      // Stop Alluxio.
+      mLocalAlluxioClusterResource.get().stopFS();
+      // Create the master using the existing journal.
+      createFileSystemMasterFromJournal();
+    } finally {
+      threadPool.shutdownNow();
```
Comment (aaudiber) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: we should wait for termination to make sure the threads aren't leaked.

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java [hunk 3]
```diff
@@ -904,4 +944,47 @@ public void exec(int depth, int concurrencyDepth, AlluxioURI path) throws Except
       }
     }
   }
+
+  class ConcurrentCreateDelete implements Callable<Void> {
```
Comment (aaudiber) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: private static?

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java (hunk 1 above)
Comment (gpang) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: done

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java (hunk 2 above)
Comment (gpang) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: done

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java (hunk 3 above)
Comment (gpang) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: I made it `private` for now, but it accesses a member of the outer class, like the other Callables here. I say we have a later PR to change them to take in a parameter, and change to `static`.

**Comment by gpang** [comment]
@aaudiber Thanks for the review! I updated the PR.

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java (hunk 3 above)
Comment (aaudiber) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: sgtm

**Pull request merged**